sync/sync.go

```diff
@@ -100,27 +101,32 @@ func (s *syncer) syncModFile(conf *config.Config, file *build.File, exitingRules
 
 	for _, req := range f.Require {
 		reqVersion := req.Mod.Version
-		var replace *modfile.Replace
+		var matchingReplace *modfile.Replace
 		for _, r := range f.Replace {
 			if r.Old.Path == req.Mod.Path {
+				matchingReplace = r
 				reqVersion = r.New.Version
-				if r.New.Path == req.Mod.Path { // we are just replacing version so don't need a replace
-					continue
-				}
-				replace = r
 			}
 		}
 
 		// Existing rule will point to the go_mod_download with the version on it so we should use the original path
-		r, ok := exitingRules[req.Mod.Path]
+		rule, ok := existingRules[req.Mod.Path]
 		if ok {
-			if replace != nil && r.Kind() == "go_repo" {
-				// Looks like we've added in a replace for this module so we need to delete the old go_repo rule
-				// and regen with a go_mod_download and a go_repo.
-				edit.RemoveTarget(file, r)
+			if matchingReplace != nil && matchingReplace.New.Path != req.Mod.Path && rule.Kind() == "go_repo" {
+				// Looks like we've added in a replace directive for this module which changes the path, so we need to
+				// delete the old go_repo rule and regenerate it with a go_mod_download and a go_repo.
+				edit.RemoveTarget(file, rule)
 			} else {
 				// Make sure the version is up-to-date
-				r.SetAttr("version", edit.NewStringExpr(reqVersion))
+				rule.SetAttr("version", edit.NewStringExpr(reqVersion))
+				// Add label for the replace directive
+				if matchingReplace != nil {
+					err := edit.AddLabel(rule, REPLACE_LABEL)
+					if err != nil {
+						return fmt.Errorf("failed to add replace label to %v: %v", req.Mod.Path, err)
```

Comment:
Is there some way we can recover from an error. Ideally we wouldn't just kill the sync because of not setting a label. It doesn't feel like the most critical to ensure is always in sync.

Reply:
I did ponder that, but this should only fail if the `labels` attribute is set to a value type that isn't list of strings.
If someone has set `labels="one-label"` or `labels=5`, it seemed like raising that as an error would be better than silently ignoring it?

What are your thoughts? I could just make it log a warning and move on, instead

Reply:
I think we should log a warning... just in case the AddLabel changes so that it errors for different reasons. I think this package should continue on any error with adding labels. Will approve with this comment though as it looks good otherwise

Reply:
Alright, switched that to a warning log now